modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -942,20 +942,88 @@ public String getAlias(String name) {
+                    if (_default.isEmpty()) {
+                        return "null";
+                    }
+
+                    List<String> final_values = _values;
+                    _default.elements().forEachRemaining((element) -> {
+                        final_values.add(element.asText());
+                    });
+                } else { // single value
+                    _values = Arrays.asList(String.valueOf(schema.getDefault()));
+                }
+
+                String defaultValue = "";
+
+                if (cp.items.isEnum) { // enum
+                    List<String> defaultValues = new ArrayList<>();
+                    for (String _value : _values) {
+                        defaultValues.add(cp.items.datatypeWithEnum + "." + toEnumVarName(_value, cp.items.dataType));
+                    }
+                    defaultValue = StringUtils.join(defaultValues, ", ");
+                } else {
+                    if (cp.items.isString) {
+                        defaultValue = StringUtils.join(_values, "\", \"");
+                        defaultValue = "\"" + defaultValue + "\"";
+                    } else {
+                        defaultValue = StringUtils.join(_values, ", ");
+                    }
+                }
+                return String.format(Locale.ROOT, "new ArrayList<>(Arrays.asList(%s))", defaultValue);
+            } else if (cp.isArray && cp.getUniqueItems()) { // set
+                // TODO
+                return null;
+            } else if (cp.isMap) { // map
+                // TODO
+                return null;
+            } else {
+                throw new RuntimeException("Error. Codegen Property must be array/set/map: " + cp);
+            }
+        } else {
+            return null;
+        }
+    }
+
     @Override
-    public String toDefaultValue(Schema schema) {
+    public String toDefaultValue(CodegenProperty cp, Schema schema) {
         schema = ModelUtils.getReferencedSchema(this.openAPI, schema);
         if (ModelUtils.isArraySchema(schema)) {
-            final String pattern;
-            if (ModelUtils.isSet(schema)) {
-                String mapInstantiationType = instantiationTypes().getOrDefault("set", "LinkedHashSet");
-                pattern = "new " + mapInstantiationType + "<%s>()";
-            } else {
-                String arrInstantiationType = instantiationTypes().getOrDefault("array", "ArrayList");
-                pattern = "new " + arrInstantiationType + "<%s>()";
+            if (schema.getDefault() == null) {
+                if (cp.isNullable) { // nullable
+                    return "null";
+                } else {
+                    final String pattern;
+                    if (ModelUtils.isSet(schema)) {
+                        String mapInstantiationType = instantiationTypes().getOrDefault("set", "LinkedHashSet");
+                        pattern = "new " + mapInstantiationType + "<%s>()";
```

Comment:
❓ Why use String concatenation here, to create a pattern that is then used with an empty argument?

The `instantiationType` should just be the variable and the `pattern` should be fixed to `new %s<%s>()` and inlined.

Reply:
I didn't write that (which was written 2 years ago).

Agreed that string concatenation can be avoided. Will try do it in the next commit.

Reply:
> then used with an empty argument?

If I guess correctly, it wasn't an empty argument before (e.g. `<String>`) but later it becomes just `<>` after we dropped JDK7 support.

Reply:
Fixed in the latest update.